### ISSUE TITLE
If the optional FOUNDRY_VERSION variable is omitted by the user, container fails to startup due to parameter not set

### ISSUE THREAD

**Issue by IcedEagle** [felddy/foundryvtt-docker#64]
When removing the FOUNDRY_VERSION environment variable, even though it's listed as optional it led to failures on start up because it was missing.

When I set up the Docker container on my Synology NAS, the UI  did set it to a default value of 0.6.5 as expected. However, I wanted to remove this variable or else tools such as Watchtower, which auto update containers, would not work properly. Watchtower would update the container, but since it recreates using the same container settings and environment variables, the Foundry software would remain at the previous version, instead of updating to the updated version as intended.

## To Reproduce

- Remove FOUNDRY_VERSION from your docker install/launch environment
- Launch the container

## Expected behavior

When lacking the FOUNDRY_VERSION variable there should be a sane default to fall back to. In this case, the version matching the image I have.

## Any helpful log output

Output when running without FOUNDRY_VERSION explicitly set by the user:

```console

Entrypoint \| 2020-08-27 02:20:34 \| [info] Starting felddy/foundryvtt container v0.6.5 | stdout
Entrypoint \| 2020-08-27 02:20:34 \| [info] Foundry Virtual Tabletop 0.6.5 is installed. | stdout
./entrypoint.sh: line 55: FOUNDRY_VERSION: parameter not set | stdout

**Comment by felddy** [comment]
Sorry this isn't working for you.  Could you attach the output from the following command to the issue.  If you are using the `alpha` or `beta` images please replace `release` with the appropriate tag.  

```bash
docker inspect --format='{{range $k, $v := .Config.Labels}}
    {{- printf "%s = \"%s\"\n" $k $v -}} {{end}}' \
    felddy/foundryvtt:release
```

Also, you post the docker command or `docker-compose.yml` file that you are using to start the container?  

## Shot in the dark 🎯 ##

I'll take a shot in the dark on the cause of the issue:  I suspect that you are setting `FOUNDRY_VERSION` to `null` or an empty string instead of not setting it at all.  If `FOUNDRY_VERSION` is left unset, it will use the value specified in the `Dockerfile` used to create the image.  

https://github.com/felddy/foundryvtt-docker/blob/6a04647ac94097d4649a74c7fd6d77187debe7f2/Dockerfile#L5

✅ This will work:
```console
❱ env -i PATH=$PATH docker run --rm --env FOUNDRY_USERNAME='me' --env FOUNDRY_PASSWORD='foobar' --publish 30000:30000/tcp --volume data:/data felddy/foundryvtt:release
Entrypoint | 2020-08-27 15:33:54 | [info] Starting felddy/foundryvtt container v0.6.5
Entrypoint | 2020-08-27 15:33:54 | [info] No Foundry Virtual Tabletop installation detected.
Entrypoint | 2020-08-27 15:33:54 | [info] Using FOUNDRY_USERNAME and FOUNDRY_PASSWORD to authenticate.
Authenticate | 2020-08-27 15:33:55 | [info] Requesting CSRF tokens from https://foundryvtt.com
Authenticate | 2020-08-27 15:33:55 | [info] Logging in as: me
Authenticate | 2020-08-27 15:33:56 | [info] Successfully logged in as: me
ReleaseURL | 2020-08-27 15:33:56 | [info] Fetching S3 pre-signed release URL for 0.6.5...
Entrypoint | 2020-08-27 15:33:56 | [info] Downloading Foundry Virtual Tabletop release.
...
Launcher | 2020-08-27 15:34:14 | [info] Starting Foundry Virtual Tabletop.
```

❌ And this will fail similarly to what you saw in your log:
```console
❱ env -i PATH=$PATH docker run --rm --env FOUNDRY_VERSION --env FOUNDRY_USERNAME='me' --env FOUNDRY_PASSWORD='foobar' --publish 30000:30000/tcp --volume data:/data felddy/foundryvtt:release
Entrypoint | 2020-08-27 15:33:39 | [info] Starting felddy/foundryvtt container v0.6.5
Entrypoint | 2020-08-27 15:33:39 | [info] No Foundry Virtual Tabletop installation detected.
Entrypoint | 2020-08-27 15:33:39 | [info] Using FOUNDRY_USERNAME and FOUNDRY_PASSWORD to authenticate.
Authenticate | 2020-08-27 15:33:39 | [info] Requesting CSRF tokens from https://foundryvtt.com
Authenticate | 2020-08-27 15:33:40 | [info] Logging in as: me
Authenticate | 2020-08-27 15:33:40 | [info] Successfully logged in as: me
./entrypoint.sh: line 75: FOUNDRY_VERSION: parameter not set

```

*Note: I'm using `env -i` to show my environment doesn't contain anything but `PATH`.*

Thanks for reporting this issue.  Let me know what you find.

**Comment by IcedEagle** [comment]
@felddy Apologies for the delay. I believe your guess was correct! Some more details which may be useful if someone else runs into the issue in the future:

- I'm using a Synology NAS with the Docker GUI to do my setup and maintenance of containers.
- I initially set up my container to have the FOUNDRY_VERSION variable. Once I decided I wanted to try it without that variable, I then shut down the container, deleted that variable from the environment variable list, and started it back up. I expect this is where Synology's Docker GUI had the variable set behind the scenes as null, as the container was created with that variable, even though from the GUI's perspective the variable was not in my environment variable list at all.
- To confirm and resolve the issue: I deleted the container, set up a new container and during setup I made sure not to include the FOUNDRY_VERSION variable. This worked correctly and I'm running 0.6.6 as expected!

Thanks again for your quick response time! I think we can close this ticket.
